instrumentation/net/http/otelhttp/internal/body_wrapper.go

```diff
@@ -0,0 +1,53 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp/internal"
+
+import (
+	"io"
+	"sync/atomic"
+)
+
+var _ io.ReadCloser = &BodyWrapper{}
+
+// BodyWrapper wraps a http.Request.Body (an io.ReadCloser) to track the number
+// of bytes read and the last error.
+type BodyWrapper struct {
+	io.ReadCloser
+	OnRead func(n int64) // must not be nil
```

Comment:
Both of these need to be defined for this to function. Can we add a "New" method to promote correct usage?